gateway/src/apicast/upstream.lua

```diff
@@ -13,27 +13,34 @@ local str_format = string.format
 local resty_resolver = require('resty.resolver')
 local resty_url = require('resty.url')
 local core_base = require('resty.core.base')
+local http_proxy = require('apicast.http_proxy')
+local str_find = string.find
+local str_sub = string.sub
 local new_tab = core_base.new_tab
 
 local _M = {
 
 }
 
 local function proxy_pass(upstream)
-    local uri = upstream.uri
-
-    return str_format('%s://%s%s%s%s',
-            uri.scheme,
-            upstream.upstream_name,
-            uri.path or ngx.var.uri or '',
-            ngx.var.is_args or '',
-            ngx.var.query_string or '')
+    return str_format('%s://%s', upstream.uri.scheme, upstream.upstream_name)
 end
 
 local mt = {
     __index = _M
 }
 
+local function split_path(path)
```

Comment:
Not something for now, but to take into account in future refactorings. I think we're already doing this in other modules, so it might be a good idea to extract methods like this one to a `Request` module that can be reused between policies.

Reply:
Yep, this is probably the most common helper function. We should offer performance optimized version and reuse it.

Reply:
Agreed.